common/sgx_workload/workload/ext_work_order_info_kme.cpp

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "crypto.h"
+#include "utils.h"
+#include "hex_string.h"
+#include "types.h"
+#include"error.h"
+
+#include "enclave_utils.h"
+#include "ext_work_order_info_kme.h"
+
+ExtWorkOrderInfoKME::ExtWorkOrderInfoKME(void) {}
+
+ExtWorkOrderInfoKME::~ExtWorkOrderInfoKME(void) {}
+
+int ExtWorkOrderInfoKME::GenerateSigningKey(
+    KeyType type, const ByteArray& nonce_hex,
+    ByteArray& signing_key, ByteArray& verification_key_hex,
+    ByteArray& verification_key_signature_hex) {
+
+    tcf_err_t result = TCF_SUCCESS;
+
+    if (type != KeyType_SECP256K1) {
+        Log(TCF_LOG_ERROR,
+            "Unsupported KeyType passed to generate signig key pair");
+        result = TCF_ERR_CRYPTO;
+        return result;
+    }
+
+    tcf::crypto::sig::PrivateKey private_sig_key;
+    tcf::crypto::sig::PublicKey public_sig_key;
+
+    try {
+        // Generate Signing key pair
+        private_sig_key.Generate();
+        public_sig_key = private_sig_key.GetPublicKey();
+
+        signing_key = StrToByteArray(private_sig_key.Serialize());
+
+        ByteArray v_key_bytes = StrToByteArray(public_sig_key.Serialize());
+        std::string v_key_hex = tcf::BinaryToHexString(v_key_bytes);
+        verification_key_hex = StrToByteArray(v_key_hex);
```

Comment:
Do we need verification_key_hex as hex string or byte array ?
If eventually we need ByteAarray can't we do this instead of multiple conversions
`ByteArray verification_key_bytes = StrToByteArray(public_sig_key.Serialize());`

If we need Hex string then
`ByteArray v_key_bytes = StrToByteArray(public_sig_key.Serialize());
std::string v_key_hex = tcf::BinaryToHexString(v_key_bytes);`

BTW Looks like as  per spec "EncryptionKeySignature" in "workerTypeData" should be hex string. But we are not assigning it as hex string and instead just converting it to base64. https://github.com/hyperledger/avalon/blob/master/tc/sgx/trusted_worker_manager/enclave/enclave_data.h#L125

Reply:
We need verificaton_key as hex bytes. Hence, i had to do multiple conversions.
To fix, the format of EncryptionKeySignatue, i will raise another PR.